include/utility/json_utility.hpp

```diff
@@ -650,5 +650,152 @@ inline bool isBackupAndRestoreRequired(const nlohmann::json& i_sysCfgJsonObj)
     }
     return false;
 }
+
+/**
+ * @brief An API to check if FRU qualifies for polling.
+ *
+ * This API checks for "pollingRequired" tag in system config JSON.
+ *
+ * @param[in] i_sysCfgJsonObj -  System config JSON object.
+ * @param[in] i_vpdFilePath - VPD file path.
+ *
+ * @return true if polling is required, false otherwise.
+ */
+inline bool isPollingRequired(const nlohmann::json& i_sysCfgJsonObj,
+                              const std::string& i_vpdFilePath)
+{
+    if (i_sysCfgJsonObj.empty() || i_vpdFilePath.empty())
+    {
+        logging::logMessage("Invalid parameters");
+        return false;
+    }
+
+    if (!i_sysCfgJsonObj.contains("frus"))
+    {
+        logging::logMessage("Missing frus section in system config JSON");
+        return false;
+    }
+
+    if (i_sysCfgJsonObj["frus"][i_vpdFilePath].at(0).contains(
+            "pollingRequired"))
+    {
+        return true;
+    }
+
+    return false;
+}
+
+/**
+ * @brief API to return parameters required for polling.
+ *
+ * This API checks for "gpioPresence" tag under "pollingRequired" in system
+ * config JSON. If present returns the parameters for GPIO polling.
+ *
+ * @throw Exceptions, if JSON does not contain "pin" and "value" tags
+ * under "gpioPresence". And std::runtime_error for invalid parameters and
+ * if JSON does not conatin "gpioPresence" and "pollingrequired" under the
+ * given VPD file path.
+ *
+ * @param[in] i_sysCfgJsonObj -  System config JSON object.
+ * @param[in] i_vpdFilePath - VPD file path.
+ *
+ * @return GPIO parameters if present, otherwise throws std::runtime_error.
+ */
+inline types::GpioPollingParameters
+    getGpioPollingParameters(const nlohmann::json& i_sysCfgJsonObj,
+                             const std::string& i_vpdFilePath)
+{
+    types::GpioPollingParameters l_pollingParameters;
+
+    if (i_sysCfgJsonObj.empty() || i_vpdFilePath.empty())
+    {
+        throw std::runtime_error("Invalid parameters");
+    }
+
+    if (!i_sysCfgJsonObj.contains("frus"))
+    {
+        throw std::runtime_error("Missing frus section in system config JSON");
+    }
+
+    if (i_sysCfgJsonObj["frus"][i_vpdFilePath].at(0).contains(
```

Comment:
As you already have API, make use of that instead.

Reply:
Updated